src/Raven.Studio/typescript/common/extensions.ts

```diff
@@ -191,9 +193,140 @@ class extensions {
+                     allBindings) => {
+
+                const activeDatabase = activeDatabaseTracker.default.database();
+                if (activeDatabase) {
+                    const bindings = allBindings();
+                    
+                    const requiredAccessLevel = ko.unwrap(valueAccessor());
+                    const strategy = bindings.requiredAccessOptions ? bindings.requiredAccessOptions.strategy : 'hide';
+                    
+                    switch (strategy) {
+                        case 'hide': {
+                            const visibleBinding = bindings.visible;
+                            const visibleValue = visibleBinding ? ko.unwrap(visibleBinding) : true;
+
+                            const hiddenBinding = bindings.enable;
+                            const hiddenValue = hiddenBinding ? ko.unwrap(hiddenBinding) : false;
+
+                            const shouldBeVisibleByKo = visibleValue && !hiddenValue;
+                            const isElementVisible = element.style.display !== "none";
+
+                            if (accessManager.default.canHandleOperation(requiredAccessLevel)()) {
+                                if (!isElementVisible && shouldBeVisibleByKo) {
+                                    element.style.display = "";
+                                }
+                            } else {
+                                if (isElementVisible) {
+                                    element.style.display = "none";
+                                }
+                            }
+                        }
+                            break;
+
+                        case 'disable': {
+                            const disableBinding = bindings.disable;
+                            const disableValue = disableBinding ? ko.unwrap(disableBinding) : false;
+
+                            const enableBinding = bindings.enable;
+                            const enableValue = enableBinding ? ko.unwrap(enableBinding) : true;
+
+                            const shouldBeEnabledByKo = !disableValue && enableValue;
+                            const isElementDisabled = element.hasAttribute("disabled");
+
+                            if (accessManager.default.canHandleOperation(requiredAccessLevel)()) {
+                                if (isElementDisabled && shouldBeEnabledByKo) {
+                                    element.setAttribute("disabled", false)
+                                }
+                            } else {
+                                if (!isElementDisabled) {
+                                    element.setAttribute("disabled", true);
+                                }
+                            }
+                        }
+                            break;
+                        
+                        case 'visibilityHidden': {
+                            // TODO
```

Comment:
throw for now:
visibilityHidden is not supported. 
or something like that.

Reply:
(optionally throw in init method)

Reply:
removed